_blog/2024-01-30-write-a-rest-api-rust.mdx

```diff
@@ -0,0 +1,262 @@
+---
+title: Writing a REST API in Rust
+description: This article talks about how you can write a Rust REST API using Axum, SQLx and Postgres. 
+author: josh
+tags: [rust, axum, guide]
+thumb: serde-rust-thumb.png 
+cover: serde-rust-thumb.png
+date: '2024-01-23T14:30:00'
+---
+
+In this article, we’re going to talk about writing a REST API in Rust! Following on from our first [Shuttle Bytes](https://www.shuttle.rs/blog/2024/01/22/introducing-shuttlebytes) stream where we live-streamed a tutorial on this, we’ve created a text write-up to allow you to follow along in text.
+
+Interested in checking out what the final code should look like? You can find it [here.](https://github.com/joshua-mo-143/shuttle-axum-example/tree/main)
+
+## Getting started
+
+### Project initialisation
+
+If you haven’t already, make sure you have Rust and Cargo installed! You can find install instructions [here.](https://www.rust-lang.org/tools/install)
+
+First of all we’ll want to initialise our project with `cargo shuttle init` (required `cargo-shuttle` to be installed). For the project name we’ll be using `shuttle-example-axum`. 
+
+Then we’ll want to set up our dependencies with the following shell snippet:
+
+```rust
+cargo add sqlx -F postgres,runtime-tokio-rustls
+cargo add shuttle-shared-db -F sqlx,postgres
+cargo add serde -F derive
+```
+
+### Adding a Database & Migrations in Rust
+
+To add a database to our project, all we need to do is to add the annotation to our main function - which allows us to automatically grab a SQLx Postgres pool:
+
+```rust
+#[shuttle_runtime::main]
+async fn main(
+    #[shuttle_shared_db::Postgres] db: PgPool,
+) -> shuttle_axum::ShuttleAxum {
+    // .. your code here
+}
+```
+
+Interested in just getting the connection string? You can disable the `sqlx` feature of `shuttle-shared-db` and instead change the type to `String`, then connect to your PgPool. You can find more information about this 
+
+When writing our migrations, since we’re using only a single migrations file we can use `include_str!()` to include the whole file’s text content as a string then execute it with the database pool.
```

Comment:
nit:
```suggestion
Since we’re using only a single migrations file for our migrations, we can use `include_str!()` to include the whole file’s text content as a string and then execute it with the database pool.
```